### PR TITLE
vm provisioning - update README and refactoring

### PR DESCRIPTION
**Pull request by giorio94**
This PR proposes an updated version of the virtual-machines provisioning README, while performing a minimal refactoring of the folder:
- Provide a brief guide on how to create, configure and export VMs with `setup-crownlabs-vm`
- Move the old scripts to a `legacy-scripts` folder since they have no longer been updated with the latest improvements
- Move the docker-related script to a `docker-scripts` folder
- Remove the templates that had not been updated (should we keep and update them?)

### CODE REVIEW

File: provisioning/virtual-machines/README.md [hunk 1]
```diff
@@ -1,81 +1,77 @@
...
-The script will install and configure:
-- **TigerVNC server**, which allows to connect to the VM desktop from a remote machine;
-- **NoVNC with websockify server**, which allows the above connection to be established through HTTP/HTTPS;
-- **Prometheus node exporter**, which exports some run-time information of the VM (e.g., CPU/memory consumption) to the Prometheus monitoring system, running on the Kubernetes cluster
-- **cloud-init**, which enables to customize some running parameters of the VM at boot time.
+Once the VM image has been correctly uploaded to the registry, it is possible to setup a new laboratory associated to it, as explained [here](../courses).
 
-To verify that the setup works, reboot the machine after running the `prepare-vm.sh` script.
-From inside the machine, start a browser and connect to page `http://localhost:6080`, using password `ccroot`.
+### TL; DR;
```
Comment (frisso) [inline provisioning/virtual-machines/README.md:13]: Marco, what does it mean "TL; DR;" here?

File: provisioning/virtual-machines/README.md [hunk 2]
```diff
@@ -1,81 +1,77 @@
...
+0. Open the `setup-crown-labs-vm` directory;
+1. Customize the basic information at the beginning of the [setup-crownlabs-vm.sh](setup-crownlabs-vm/setup-crownlabs-vm.sh)
+   script (e.g. xubuntu version and docker registry configuration);
```
Comment (frisso) [inline provisioning/virtual-machines/README.md:16]: Remove new line here

File: provisioning/virtual-machines/README.md [hunk 3]
```diff
@@ -1,81 +1,77 @@
...
+6. Execute `./setup-crownlabs-vm <vm-name> export crownlabs` to export the VM.
 
-### Conversion to raw format
+Please, refer to the corresponding [README file](setup-crownlabs-vm/README.md) for additional information about this script.
 
-Once you made sure that the VM has been properly configured and runs smothly, shutdown again the VM and convert it to the `qcow2` format, which is used by the Kubernetes virtualization module (Kube-virt).
-This can be done with the [convert-vm.sh](scripts/convert-vm.sh) script, typing the following command:
+Some predefined playbook are available in the [ansible](setup-crownlabs-vm/ansible) folder (those named `crownlabs` take care of installing the tools required by CrownLabs).
+In case you want to install different software, you can simply create a new
+ansible script, and optionally new ansible roles.
```
Comment (frisso) [inline provisioning/virtual-machines/README.md:27]: Remove new line here

File: provisioning/virtual-machines/README.md [hunk 4]
```diff
@@ -1,81 +1,77 @@
...
+- **TigerVNC server**, which allows to connect to the VM desktop from a remote machine;
+- **NoVNC with websockify server**, which allows the above connection to be established through HTTP/HTTPS;
+- **Prometheus node exporter**, which exports some run-time information of the VM (e.g., CPU/memory consumption) to the Prometheus monitoring system, running on the Kubernetes cluster
+- **cloud-init**, which enables to customize some running parameters of the VM at boot time.
 
-```sh
-$ docker push registry.crown-labs.ipv6.polito.it/<image_name>:latest
-```
 
 ## Cleaning up private registry
```
Comment (frisso) [inline provisioning/virtual-machines/README.md:70]: ==>  ## Cleaning up private Docker registry

File: provisioning/virtual-machines/README.md (hunk 1 above)
Comment (giorio94) [reply provisioning/virtual-machines/README.md]: It should refer to the brief version of a long description. I replaced it with "Quick start guide".

**Comment by giorio94** [comment]
I should have fixed the identified problems

**Pull request merged**